security/security-core/src/main/java/io/camunda/security/configuration/OidcAuthenticationConfiguration.java

```diff
@@ -22,6 +22,7 @@ public class OidcAuthenticationConfiguration {
   private String authorizationUri;
   private Set<String> audiences;
   private String usernameClaim = "sub";
+  private String applicationIdClaim;
```

Comment:
:wrench: We should probably validate that it's not the same claim as `usernameClaim`

Reply:
Would there be a strong benefit to validating that here? Or the other way around, what would be the negatives of not validating it?

Reply:
Yea, as discussed in the meeting this afternoon, you could configure these and then we'd look at the token and if there is a username claim we'd have preference for that, otherwise its application ID. for now I'll leave off the validation as we should really document the intentions

Reply:
Hm, I'm still in favor of a validation because setting them both to the same claim key effectively disables applications entirely (username always has precedence). It's just not a valid configuration and we should validate against it if possible.

Reply:
Yes, we have the `ConfigurationEntry` interface that we use to validate this sort of thing.